src/services/Tx.ts

```diff
@@ -66,6 +97,78 @@ const createPool = async (
   )
 }
 
+const sellAsset = async (
+  api: ApiPromise,
+  address: string,
+  soldAssetId: string,
+  boughtAssetId: string,
+  amount: BN,
+  minAmountOut: BN
+): Promise<void> => {
+  signTx(
+    api,
+    api.tx.xyk.sellAsset(soldAssetId, boughtAssetId, amount, minAmountOut),
+    address,
+    await Query.getNonce(api, address)
```

Comment:
What happens if for new account no nonce is specified? 
Logic of nonce management in https://github.com/mangata-finance/mangata-sdk/blob/develop/src/services/Tx.ts#L23 is complicated and confusing:
if nonce is provided, it is yet not guaranteed provided nonce will be used. 
The logic should be altered to use either a nonce provided as an argument, or nonce provided by nonce manager according to the algorithm.